File: backend/app/models/appeal.py
from datetime import datetime
from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship
from app.db.base_class import Base

class Appeal(Base):
    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, ForeignKey("task.id"), nullable=False)
    creator_id = Column(Integer, ForeignKey("user.id"), nullable=False)
    reason = Column(Text, nullable=False)
    status = Column(String(20), default="pending")  # 'pending', 'processing', 'resolved', 'rejected'
    admin_reply = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    task = relationship("Task")
    creator = relationship("User")

File: frontend/src/api/appeal.ts
/**
 * Appeal API 接口
 */

export interface Appeal {
  id: number
  taskId: number
  creatorId: number
  reason: string
  status: 'pending' | 'processing' | 'resolved' | 'rejected'
  adminReply?: string
  createdAt: string
  updatedAt: string
  task?: {
    id: number
    title: string
  }
}

export interface CreateAppealPayload {
  taskId: number
  reason: string
}

// 获取申诉列表
export async function getAppeals(): Promise<{ success: boolean; message: string; data?: Appeal[] }> {
  const response = await fetch('/api/appeal/my', {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${localStorage.getItem('token')}`
    }
  })

  const result = await response.json()

  if (!response.ok) {
    throw new Error(result.message || '获取申诉列表失败')
  }

  return result
}

// 创建申诉
export async function createAppeal(payload: CreateAppealPayload): Promise<{ success: boolean; message: string; data?: Appeal }> {
  const response = await fetch('/api/appeal/create', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${localStorage.getItem('token')}`
    },
    body: JSON.stringify(payload)
  })

  const result = await response.json()

  if (!response.ok) {
    throw new Error(result.message || '创建申诉失败')
  }

  return result
}

// 获取申诉详情
export async function getAppealDetail(appealId: number): Promise<{ success: boolean; message: string; data?: Appeal }> {
  const response = await fetch(`/api/appeal/${appealId}`, {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${localStorage.getItem('token')}`
    }
  })

  const result = await response.json()

  if (!response.ok) {
    throw new Error(result.message || '获取申诉详情失败')
  }

  return result
}
